portal-impl/src/com/liferay/portlet/internal/PortletBagImpl.java

```diff
@@ -303,21 +307,55 @@ public void setPortletName(String portletName) {
 		_portletName = portletName;
 	}
 
-	private final <T> List<T> _getList(Class<?> clazz) {
+	private final <T> List<T> _getList(Class<T> clazz) {
 		ServiceTrackerList<Class<?>> serviceTrackerList =
 			_serviceTrackerListMap.computeIfAbsent(
 				clazz,
 				key ->
 					(ServiceTrackerList<Class<?>>)
 						(ServiceTrackerList)ServiceTrackerListFactory.open(
-							_bundleContext, clazz, _filterString));
+							_bundleContext, clazz, _filterString,
+							new ServiceTrackerCustomizer<T, T>() {
+
+								@Override
+								public T addingService(
+									ServiceReference<T> serviceReference) {
+
+									long companyId = GetterUtil.getLong(
+										serviceReference.getProperty(
+											"com.liferay.portlet.company"),
+										CompanyConstants.SYSTEM);
+
+									if (companyId != _companyId) {
+										return null;
+									}
```

Comment:
This makes portlets registered for SYSTEM company to work as expected right?

Reply:
The previous code:

```java
	long companyId = GetterUtil.getLong(
		serviceReference.getProperty(
			"com.liferay.portlet.company"),
		CompanyConstants.SYSTEM);
```

tries to get a property named `com.liferay.portlet.company` which, unless explicitly set (like in CXs), is usually missing. Thus, it will be `CompanyConstants.SYSTEM` for most of the portlets, which is the expected value since currently, we are holding all porlets inside that company unless they are registered for a specific one (which only happens for CXs).

Reply:
> we are holding all porlets inside that company unless they are registered for a specific one

That happens because we substituted the [maps in `PortletLocalServiceImpl`](https://github.com/liferay-frontend/liferay-portal/pull/3338/files#diff-b616dab9f6e1810a2fa02b1f41c0783a708546810989fc34973a5c6491728fe5L2770) by a `CompanyPortletMap` which first looks for things in the active (ambient) company, then [falls back to the system one](https://github.com/liferay-frontend/liferay-portal/pull/3338/files#diff-8d6cd4b6886f9473cf5894a961a99b78f832f3f63828edb609974cace0bb2c7bR51).

Thus, it's enough to register all entities without `com.liferay.portlet.company` under the system company to make them visible to all existing companies.